src/core/lib/transport/metadata.c

```diff
@@ -504,7 +507,8 @@ static void grow_mdtab(mdtab_shard *shard) {
 }
 
 static void rehash_mdtab(mdtab_shard *shard) {
-  if (shard->free > shard->capacity / 4) {
+  if ((size_t)gpr_atm_no_barrier_load(&shard->free_estimate) >
```

Comment:
I believe it is possible for free_estimate to transiently drop below zero, and in that case you'll make the wrong decision here because you're casting to an unsigned value.